invoke.go

```diff
@@ -22,16 +22,45 @@ package dig
 
 import (
 	"fmt"
-	"reflect"
-
 	"go.uber.org/dig/internal/digreflect"
 	"go.uber.org/dig/internal/graph"
+	"reflect"
 )
 
-// An InvokeOption modifies the default behavior of Invoke. It's included for
-// future functionality; currently, there are no concrete implementations.
+// An InvokeOption modifies the default behavior of Invoke.
 type InvokeOption interface {
-	unimplemented()
+	applyInvokeOption(*invokeOptions)
+}
+
+type invokeOptions struct {
+	Info *InvokeInfo
+}
+
+// InvokeInfo provides information about an Invoke.
+type InvokeInfo struct {
+	Inputs []*Input
+}
+
+// FillInvokeInfo is an InvokeOption that writes info on the types requested
+// by the Invoke function into the specified InvokeInfo.
+// For example:
+//
+//	var info dig.InvokeInfo
+//	err := c.Invoke(..., dig.FillInvokeInfo(&info))
```

Comment:
Maybe provide a _real_ example of a function, and what the corresponding provided info looks like?